# Covid-19 Project: Data preparation (part 2)

In [1]:
# Import packages

import pandas as pd
import numpy as np

Load datasets prerpared in previous steps <br>
See Covid-19Project_DataPreparation1

In [2]:
# Cumulative cases
confirmed_global_cleaned = pd.read_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/confirmed_global_cleaned.csv")
deaths_global_cleaned = pd.read_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/deaths_global_cleaned.csv")
recovered_global_cleaned = pd.read_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/recovered_global_cleaned.csv")

# Cases per day
confirmed_global_cpr = pd.read_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/confirmed_global_cpr.csv")
deaths_global_cpr = pd.read_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/deaths_global_cpr.csv")
recovered_global_cpr = pd.read_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/recovered_global_cpr.csv")

Small adjustments of loaded datasets:

In [3]:
# df = "..._cleaned" dataset

def preparation2 (df):

    df.set_index("Country/Region", inplace=True)
    df.index.rename("Country", inplace=True)
    
    return df

In [4]:
confirmed_global_cleaned2 = preparation2(confirmed_global_cleaned)
confirmed_global_cleaned2.head()

1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  1/28/20  \
Country                                                                      
Afghanistan        0        0        0        0        0        0        0   
Albania            0        0        0        0        0        0        0   
Algeria            0        0        0        0        0        0        0   
Andorra            0        0        0        0        0        0        0   
Angola             0        0        0        0        0        0        0   

             1/29/20  1/30/20  1/31/20  ...  1/29/21  1/30/21  1/31/21  \
Country                                 ...                              
Afghanistan        0        0        0  ...    54939    55008    55023   
Albania            0        0        0  ...    76350    77251    78127   
Algeria            0        0        0  ...   106887   107122   107339   
Andorra            0        0        0  ...     9837     9885     9937   
Angola             0        0        0  ...    19723    19782    19796   

             2/1/21  2/2/21  2/3/21  2/4/21  2/5/21  2/6/21  2/7/21  
Country                                                              
Afghanistan   55059   55121   55174   55231   55265   55330   55335  
Albania       78992   79934   80941   81993   83082   84212   85336  
Algeria      107578  107841  108116  108381  108629  108629  109088  
Andorra        9972   10017   10070   10137   10172   10206   10251  
Angola        19829   19900   19937   19996   20030   20062   20086  

[5 rows x 383 columns]

In [5]:
recovered_global_cleaned2 = preparation2(recovered_global_cleaned)
recovered_global_cleaned2.tail()

1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  \
Country                                                                    
Wallis and Futuna         0        0        0        0        0        0   
West Bank and Gaza        0        0        0        0        0        0   
Yemen                     0        0        0        0        0        0   
Zambia                    0        0        0        0        0        0   
Zimbabwe                  0        0        0        0        0        0   

                    1/28/20  1/29/20  1/30/20  1/31/20  ...  1/29/21  1/30/21  \
Country                                                 ...                     
Wallis and Futuna         0        0        0        0  ...        4        4   
West Bank and Gaza        0        0        0        0  ...   148140   148583   
Yemen                     0        0        0        0  ...     1425     1426   
Zambia                    0        0        0        0  ...    44133    45443   
Zimbabwe                  0        0        0        0  ...    24872    25361   

                    1/31/21  2/1/21  2/2/21  2/3/21  2/4/21  2/5/21  2/6/21  \
Country                                                                       
Wallis and Futuna         4       4       4       4       4       4       4   
West Bank and Gaza   148993  149621  150123  150663  151082  151690  152120   
Yemen                  1426    1426    1427    1427    1427    1427    1428   
Zambia                48000   49394   49696   50479   51305   52045   53410   
Zimbabwe              26044   26583   26794   27391   27759   28145   28383   

                    2/7/21  
Country                     
Wallis and Futuna        4  
West Bank and Gaza  152579  
Yemen                 1428  
Zambia               54835  
Zimbabwe             28551  

[5 rows x 383 columns]

In [6]:
deaths_global_cleaned2 = preparation2(deaths_global_cleaned)
deaths_global_cleaned2.head()

1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  1/28/20  \
Country                                                                      
Afghanistan        0        0        0        0        0        0        0   
Albania            0        0        0        0        0        0        0   
Algeria            0        0        0        0        0        0        0   
Andorra            0        0        0        0        0        0        0   
Angola             0        0        0        0        0        0        0   

             1/29/20  1/30/20  1/31/20  ...  1/29/21  1/30/21  1/31/21  \
Country                                 ...                              
Afghanistan        0        0        0  ...     2399     2400     2400   
Albania            0        0        0  ...     1358     1369     1380   
Algeria            0        0        0  ...     2884     2888     2891   
Andorra            0        0        0  ...      100      101      101   
Angola             0        0        0  ...      464      464      466   

             2/1/21  2/2/21  2/3/21  2/4/21  2/5/21  2/6/21  2/7/21  
Country                                                              
Afghanistan    2404    2405    2406    2407    2407    2408    2410  
Albania        1393    1398    1404    1418    1433    1446    1460  
Algeria        2894    2898    2900    2904    2909    2909    2914  
Andorra         101     102     103     103     104     105     106  
Angola          466     468     470     473     474     475     475  

[5 rows x 383 columns]

In [7]:
confirmed_global_cpr.tail()

Unnamed: 0  Afghanistan  Albania  Algeria  Andorra  Angola  Anguilla  \
378     2/3/21           53     1007      275       53      37         0   
379     2/4/21           57     1052      265       67      59         0   
380     2/5/21           34     1089      248       35      34         0   
381     2/6/21           65     1130        0       34      32         0   
382     2/7/21            5     1124      459       45      24         0   

     Antigua and Barbuda  Argentina  Armenia  ...  Uruguay  Uzbekistan  \
378                    0       9196      190  ...      548          58   
379                   19       8891      147  ...      589          57   
380                    9       8374      158  ...      499          73   
381                   11       6680      211  ...      509          56   
382                   11       3658      151  ...      499          53   

     Vanuatu  Venezuela  Vietnam  Wallis and Futuna  West Bank and Gaza  \
378        0        563       66                  0                 470   
379        0        460        9                  0                 661   
380        0        456       19                  0                 472   
381        0        412        9                  0                 470   
382        0        473       16                  0                 572   

     Yemen  Zambia  Zimbabwe  
378      0    1256       150  
379      0    1514       207  
380      2    1424       160  
381      3    1000       156  
382      0    1206        65  

[5 rows x 220 columns]

In [8]:
# df_cpr = "..._cpr" dataset

def preparation2_cpr (df_cpr):
    
    df_cpr.set_index("Unnamed: 0", inplace=True)
    df_cpr.index.name = None
    df_cpr_trans = df_cpr.transpose()
    df_cpr_trans.index.rename("Country", inplace=True)
    df_cpr_trans.sort_index(inplace=True)
    return df_cpr_trans


In [9]:
confirmed_global_cpr2 = preparation2_cpr(confirmed_global_cpr)
confirmed_global_cpr2.head()

1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  1/28/20  \
Country                                                                      
Afghanistan        0        0        0        0        0        0        0   
Albania            0        0        0        0        0        0        0   
Algeria            0        0        0        0        0        0        0   
Andorra            0        0        0        0        0        0        0   
Angola             0        0        0        0        0        0        0   

             1/29/20  1/30/20  1/31/20  ...  1/29/21  1/30/21  1/31/21  \
Country                                 ...                              
Afghanistan        0        0        0  ...       48       69       15   
Albania            0        0        0  ...      896      901      876   
Algeria            0        0        0  ...      277      235      217   
Andorra            0        0        0  ...       58       48       52   
Angola             0        0        0  ...       51       59       14   

             2/1/21  2/2/21  2/3/21  2/4/21  2/5/21  2/6/21  2/7/21  
Country                                                              
Afghanistan      36      62      53      57      34      65       5  
Albania         865     942    1007    1052    1089    1130    1124  
Algeria         239     263     275     265     248       0     459  
Andorra          35      45      53      67      35      34      45  
Angola           33      71      37      59      34      32      24  

[5 rows x 383 columns]

In [10]:
deaths_global_cpr2 = preparation2_cpr(deaths_global_cpr)
deaths_global_cpr2.head()

1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  1/28/20  \
Country                                                                      
Afghanistan        0        0        0        0        0        0        0   
Albania            0        0        0        0        0        0        0   
Algeria            0        0        0        0        0        0        0   
Andorra            0        0        0        0        0        0        0   
Angola             0        0        0        0        0        0        0   

             1/29/20  1/30/20  1/31/20  ...  1/29/21  1/30/21  1/31/21  \
Country                                 ...                              
Afghanistan        0        0        0  ...        2        1        0   
Albania            0        0        0  ...        8       11       11   
Algeria            0        0        0  ...        3        4        3   
Andorra            0        0        0  ...        0        1        0   
Angola             0        0        0  ...        0        0        2   

             2/1/21  2/2/21  2/3/21  2/4/21  2/5/21  2/6/21  2/7/21  
Country                                                              
Afghanistan       4       1       1       1       0       1       2  
Albania          13       5       6      14      15      13      14  
Algeria           3       4       2       4       5       0       5  
Andorra           0       1       1       0       1       1       1  
Angola            0       2       2       3       1       1       0  

[5 rows x 383 columns]

In [11]:
recovered_global_cpr2 = preparation2_cpr(recovered_global_cpr)
recovered_global_cpr2.head()

1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  1/28/20  \
Country                                                                      
Afghanistan        0        0        0        0        0        0        0   
Albania            0        0        0        0        0        0        0   
Algeria            0        0        0        0        0        0        0   
Andorra            0        0        0        0        0        0        0   
Angola             0        0        0        0        0        0        0   

             1/29/20  1/30/20  1/31/20  ...  1/29/21  1/30/21  1/31/21  \
Country                                 ...                              
Afghanistan        0        0        0  ...       23        3       70   
Albania            0        0        0  ...      514      507      497   
Algeria            0        0        0  ...      211      196      192   
Andorra            0        0        0  ...       44       59       48   
Angola             0        0        0  ...      226       97       20   

             2/1/21  2/2/21  2/3/21  2/4/21  2/5/21  2/6/21  2/7/21  
Country                                                              
Afghanistan      44      75     184      13      23       9      14  
Albania         498     455     581     585     676     656     569  
Algeria         186     202     208     196     194       0     362  
Andorra         113      46      61      82      49      48      41  
Angola          145      52     103      46      50      44      65  

[5 rows x 383 columns]

<b> Prepare dataset with population to calculate the incidence (7-days) rate </b> 
<br>
<br>
Updated population data (2020) was scraped from: https://www.worldometers.info/world-population/population-by-country/ <br>
See: Covid-19Project_DataPreparation2A_PopulationWebScrap

In [12]:
# Load population dataset

population = pd.read_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/population2020_scrap.csv")
population.set_index("Country", inplace=True)
population.shape

(235, 1)

In [13]:
population.info()

<class 'pandas.core.frame.DataFrame'>
Index: 235 entries, Afghanistan to Zimbabwe
Data columns (total 1 columns):
 #   Column             Non-Null Count  Dtype
---  ------             --------------  -----
 0   Population (2020)  235 non-null    int64
dtypes: int64(1)
memory usage: 3.7+ KB


In [14]:
# Create dataframe with country names used in the Covid-19 dataset

countries_dataset = confirmed_global_cleaned2.index.values
countries_dataset = pd.DataFrame([countries_dataset], index=["Country"]).transpose()
countries_dataset.set_index("Country", inplace=True)

# Column for further steps - to compare both datasets more easily
#countries_dataset["total"] = confirmed_global_cpr2["total"]
countries_dataset.head()
countries_dataset.shape

(219, 0)

The population dataset contains 235 countries, while Covid-19 dataset included 219 countries <br>
It means both datasets need to merged using "inner" join, so that only countries from the Covid19 dataset are included

In [15]:
merged = pd.concat([countries_dataset, population], axis=1, join="inner")
merged.head()

Population (2020)
Country                       
Afghanistan           38928346
Albania                2877797
Algeria               43851044
Andorra                  77265
Angola                32866272

In [16]:
merged.shape

(196, 1)

Merged dataset contains 196 countries, instead of 219 -> 23 country names probably vary or some countries are missing <br>
Let's check it:

In [17]:
# countries_dataset.compare(population2, align_axis=0) -> from version 1.1.0, here 1.0.5 and cannot be upgraded in Anaconda

In [18]:
# This time both datasets will be merged using outer join and
# and then it will be checked for where the population values are missing (NaN)

test = pd.concat([countries_dataset, merged], axis=1, join="outer")
test[test["Population (2020)"].isna()]

Population (2020)
Bonaire, Sint Eustatius and Saba                NaN
Burma                                           NaN
Congo (Brazzaville)                             NaN
Congo (Kinshasa)                                NaN
Cote d'Ivoire                                   NaN
Curacao                                         NaN
Czechia                                         NaN
Diamond Princess                                NaN
Falkland Islands (Malvinas)                     NaN
Faroe Islands                                   NaN
Korea, South                                    NaN
Kosovo                                          NaN
MS Zaandam                                      NaN
Reunion                                         NaN
Saint Kitts and Nevis                           NaN
Saint Pierre and Miquelon                       NaN
Saint Vincent and the Grenadines                NaN
Sao Tome and Principe                           NaN
St Martin                                       NaN
Taiwan*                                         NaN
Turks and Caicos Islands                        NaN
Wallis and Futuna                               NaN
West Bank and Gaza                              NaN

MS Zaandam and Diamond Princess are names of cruise ships <br>
In other cases, it is very likely that the country names just vary (e.g. "and", "Burma", etc.) <br>

In [19]:
# Check country names in "population" dataset

population.index.values

array(['Afghanistan', 'Albania', 'Algeria', 'American Samoa', 'Andorra',
       'Angola', 'Anguilla', 'Antigua and Barbuda', 'Argentina',
       'Armenia', 'Aruba', 'Australia', 'Austria', 'Azerbaijan',
       'Bahamas', 'Bahrain', 'Bangladesh', 'Barbados', 'Belarus',
       'Belgium', 'Belize', 'Benin', 'Bermuda', 'Bhutan', 'Bolivia',
       'Bosnia and Herzegovina', 'Botswana', 'Brazil',
       'British Virgin Islands', 'Brunei', 'Bulgaria', 'Burkina Faso',
       'Burundi', 'Cabo Verde', 'Cambodia', 'Cameroon', 'Canada',
       'Caribbean Netherlands', 'Cayman Islands',
       'Central African Republic', 'Chad', 'Channel Islands', 'Chile',
       'China', 'Colombia', 'Comoros', 'Congo', 'Cook Islands',
       'Costa Rica', 'Croatia', 'Cuba', 'Curaçao', 'Cyprus',
       'Czech Republic (Czechia)', "Côte d'Ivoire", 'DR Congo', 'Denmark',
       'Djibouti', 'Dominica', 'Dominican Republic', 'Ecuador', 'Egypt',
       'El Salvador', 'Equatorial Guinea', 'Eritrea', 'Estonia',
       'Esw

In [20]:
# Rename country names

# Population data for Bonaire, Sint Eustatius and Saba & Kosovo is not given

population_renamed = population.rename(index={"Congo": "Congo (Brazzaville)",                
                                              "Côte d'Ivoire": "Cote d'Ivoire", 
                                              "Curaçao": "Curacao", 
                                              "Czech Republic (Czechia)": "Czechia", 
                                              "DR Congo":"Congo (Kinshasa)",
                                              "Falkland Islands": "Falkland Islands (Malvinas)", 
                                              "Faeroe Islands": "Faroe Islands", 
                                                "Myanmar": "Burma",
                                              "Réunion" : "Reunion",
                                              "Sao Tome & Principe":"Sao Tome and Principe", 
                                              "Saint Kitts & Nevis" : "Saint Kitts and Nevis", 
                                              "Saint Martin": "St Martin", 
                                              "Saint Pierre & Miquelon": "Saint Pierre and Miquelon",
                                              "St. Vincent & Grenadines":"Saint Vincent and the Grenadines",
                                              "South Korea": "Korea, South", 
                                              "State of Palestine": "West Bank and Gaza",
                                              "Taiwan": "Taiwan*", 
                                              "Turks and Caicos": "Turks and Caicos Islands", 
                                              "Wallis & Futuna": "Wallis and Futuna", 
                                             })

Create complete population dataset:

In [21]:
population_complete = pd.concat([countries_dataset, population_renamed], axis=1, join="inner")
population_complete.rename(columns={"Population (2020)": "Population"}, inplace=True)
population_complete.info()

<class 'pandas.core.frame.DataFrame'>
Index: 215 entries, Afghanistan to Zimbabwe
Data columns (total 1 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   Population  215 non-null    int64
dtypes: int64(1)
memory usage: 3.4+ KB


In [22]:
# Number of entries is correct -> originial countries (covid-19) dataset: 219 entries 
# 4 items are missing:
# a) in the "population" dataset Kosovo, Bonaire, Sint Eustatius and Saba were not included
# b) MS Zaandam & Diamond Princess are cruise ships

Population data for Kosovo can be extracted from the World Bank Dataset, however that will be data from 2019
<br>
<br>
Source: The World Bank Population dataset (data until 2019) https://data.worldbank.org/indicator/SP.POP.TOTL 

In [23]:
# Load World Bank dataset
population_WB_2019 = pd.read_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/API_SP.POP.TOTL_DS2_en_csv_v2_1976634.csv", skiprows =4)

In [24]:
population_WB_2019.head()

Country Name Country Code     Indicator Name Indicator Code       1960  \
0        Aruba          ABW  Population, total    SP.POP.TOTL    54211.0   
1  Afghanistan          AFG  Population, total    SP.POP.TOTL  8996973.0   
2       Angola          AGO  Population, total    SP.POP.TOTL  5454933.0   
3      Albania          ALB  Population, total    SP.POP.TOTL  1608800.0   
4      Andorra          AND  Population, total    SP.POP.TOTL    13411.0   

        1961       1962       1963       1964       1965  ...        2012  \
0    55438.0    56225.0    56695.0    57032.0    57360.0  ...    102560.0   
1  9169410.0  9351441.0  9543205.0  9744781.0  9956320.0  ...  31161376.0   
2  5531472.0  5608539.0  5679458.0  5735044.0  5770570.0  ...  25107931.0   
3  1659800.0  1711319.0  1762621.0  1814135.0  1864791.0  ...   2900401.0   
4    14375.0    15370.0    16412.0    17469.0    18549.0  ...     82427.0   

         2013        2014        2015        2016        2017        2018  \
0    103159.0    103774.0    104341.0    104872.0    105366.0    105845.0   
1  32269589.0  33370794.0  34413603.0  35383128.0  36296400.0  37172386.0   
2  26015780.0  26941779.0  27884381.0  28842484.0  29816748.0  30809762.0   
3   2895092.0   2889104.0   2880703.0   2876101.0   2873457.0   2866376.0   
4     80774.0     79213.0     78011.0     77297.0     77001.0     77006.0   

         2019  2020  Unnamed: 65  
0    106314.0   NaN          NaN  
1  38041754.0   NaN          NaN  
2  31825295.0   NaN          NaN  
3   2854191.0   NaN          NaN  
4     77142.0   NaN          NaN  

[5 rows x 66 columns]

In [25]:
# Prepare dataset

# Extract only columns that are necessary
population_WB_2019_short = population_WB_2019[["Country Name", "2019"]]
population_WB_2019_short.set_index("Country Name", inplace=True)
population_WB_2019_short.head()

2019
Country Name            
Aruba           106314.0
Afghanistan   38041754.0
Angola        31825295.0
Albania        2854191.0
Andorra          77142.0

In [26]:
# Select only data for Kosovo

population_WB_2019_Kosovo = population_WB_2019_short.loc["Kosovo"]

In [29]:
# Convert into dataframe

population_WB_2019_Kosovo2 = pd.DataFrame([population_WB_2019_Kosovo])
population_WB_2019_Kosovo2

2019
Kosovo  1794248.0

In [30]:
# Add Kosovo to "population_complete" dataframe

population_WB_2019_Kosovo2.rename(columns={"2019": "Population"}, inplace=True)
population_complete_Kosovo = pd.concat([population_complete, population_WB_2019_Kosovo2])
population_complete_Kosovo.tail()

Population
West Bank and Gaza   5101414.0
Yemen               29825964.0
Zambia              18383955.0
Zimbabwe            14862924.0
Kosovo               1794248.0

In [31]:
# Sort "population_complete_Kosovo by" index

population_complete_Kosovo.sort_index(inplace=True)
population_complete_Kosovo.shape

(216, 1)

<b> Calculate incidence (7-days) rate </b><br>
<br>
In Germany: Incidence rate is a sum of daily (new) cases from the last 7 days per 100 000 inhabitants <br>
This rate is crucial for many decisions the government makes nowadays, e.g. whether a country is classified as a risk region or not

In [32]:
# Divide the population through 100 000 to get results "per 100 000 inhabitants"

# Dataset without Kosovo
population_complete["div100"] = (population_complete["Population"] / 100000).round(2)
population_complete.head()

Population  div100
Country                        
Afghanistan    38928346  389.28
Albania         2877797   28.78
Algeria        43851044  438.51
Andorra           77265    0.77
Angola         32866272  328.66

In [33]:
# Dataset with Kosovo
population_complete_Kosovo["div100"] = (population_complete_Kosovo["Population"] / 100000).round(2)
population_complete_Kosovo.head()

Population  div100
Afghanistan  38928346.0  389.28
Albania       2877797.0   28.78
Algeria      43851044.0  438.51
Andorra         77265.0    0.77
Angola       32866272.0  328.66

In [34]:
# Save population datasets to csv-files

population_complete.to_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/population_complete.csv")
population_complete_Kosovo.to_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/population_complete_Kosovo.csv")

In [35]:
# Calculate 7-days incidence rate

# df = dataset with cases per day ("..._cpr2")

def incidence (df):
    
    # merge cpr dataset with population, only column "div100"
    df_global_population = pd.concat([df, population_complete_Kosovo["div100"]], axis=1, join="inner")
    
    # Calculate the sum of the last 7 days
    df_global_population["sum7days"] = df_global_population.iloc[:,-8:-1].sum(axis=1)
    
    # Calculate 7-days incidence
    df_global_population["incidence"] = (df_global_population["sum7days"] / df_global_population["div100"]).round(2)
    
    return df_global_population


#df["sum7days"] = df.iloc[:,-7:].sum(axis=1)


In [36]:
confirmed_global_incidence = incidence(confirmed_global_cpr2)
confirmed_global_incidence.head()

1/22/20  1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  1/28/20  \
Afghanistan        0        0        0        0        0        0        0   
Albania            0        0        0        0        0        0        0   
Algeria            0        0        0        0        0        0        0   
Andorra            0        0        0        0        0        0        0   
Angola             0        0        0        0        0        0        0   

             1/29/20  1/30/20  1/31/20  ...  2/1/21  2/2/21  2/3/21  2/4/21  \
Afghanistan        0        0        0  ...      36      62      53      57   
Albania            0        0        0  ...     865     942    1007    1052   
Algeria            0        0        0  ...     239     263     275     265   
Andorra            0        0        0  ...      35      45      53      67   
Angola             0        0        0  ...      33      71      37      59   

             2/5/21  2/6/21  2/7/21  div100  sum7days  incidence  
Afghanistan      34      65       5  389.28       312       0.80  
Albania        1089    1130    1124   28.78      7209     250.49  
Algeria         248       0     459  438.51      1749       3.99  
Andorra          35      34      45    0.77       314     407.79  
Angola           34      32      24  328.66       290       0.88  

[5 rows x 386 columns]

In [37]:
# Save the dataset with incidence rates to a csv file:

confirmed_global_incidence.to_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/confirmed_global_incidence.csv")

<b> Table with coordinates - preparation for further steps </b>

In [38]:
# Drop duplicates (China, Australia, Canada) and change coordinates

coord = pd.read_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/coordinates.csv")

# Drop duplicates for Australia, Canada and China
# Coordinates for Australia = Australian Capital Territory; Canada = Alberta; China = Anhui

coord.drop_duplicates(subset="Country/Region", keep="first", inplace=True)
coord.drop(columns="Province/State", inplace=True)
coord.set_index("Country/Region", inplace=True )
coord.index.rename("Country", inplace=True)
coord.rename(index={"US": "United States"}, inplace=True)
coord.sort_index(inplace=True)
coord.tail()

#coord.to_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/coordinates_prep.csv")

Lat        Long
Country                                  
Wallis and Futuna  -14.293800 -178.116500
West Bank and Gaza  31.952200   35.233200
Yemen               15.552727   48.516388
Zambia             -13.133897   27.849332
Zimbabwe           -19.015438   29.154857

In [39]:
# Save to csv-file

coord.to_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/coordinates_prep.csv")

In [40]:
# Coordinates for incidence dataset

coord_population = pd.concat([coord, population_complete_Kosovo], axis=1, join="inner")
coord_population.info()

<class 'pandas.core.frame.DataFrame'>
Index: 216 entries, Afghanistan to Zimbabwe
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Lat         216 non-null    float64
 1   Long        216 non-null    float64
 2   Population  216 non-null    float64
 3   div100      216 non-null    float64
dtypes: float64(4)
memory usage: 8.4+ KB


In [41]:
coord_population.drop(columns=["Population", "div100"], inplace=True)
coord_population.head()

Lat       Long
Afghanistan  33.93911  67.709953
Albania      41.15330  20.168300
Algeria      28.03390   1.659600
Andorra      42.50630   1.521800
Angola      -11.20270  17.873900

In [42]:
# Save to csv-file
coord_population.to_csv("C:/Users/Agnieszka/Downloads/Datasets/Covid_temp/coordinates_for_incidence.csv")

<b> Created datasets (csv): </b> <br> <br>
confirmed_global_incidence <br> 
coordinates_prep <br> 
coordinates_for_incidence <br> 
population_complete <br> 
population_complete_Kosovo <br> 